src/cmds/restic/cmd_snapshots.go

```diff
@@ -121,5 +136,40 @@ func runSnapshots(opts SnapshotOptions, gopts GlobalOptions, args []string) erro
 
 	tab.Write(os.Stdout)
 
+	return
+}
+
+//Snapshot provides struct to store snapshots
+type Snapshot struct {
+	ID        string   `json:"id"`
+	Date      string   `json:"date"`
+	Host      string   `json:"host"`
+	Tags      []string `json:"tags"`
+	Directory []string `json:"directory"`
+}
+
+//jsonoutput provides machine redability
+func jsonoutput(list []*restic.Snapshot) error {
+	var response []Snapshot
+
+	for _, sn := range list {
+
+		k := Snapshot{
+			ID:        sn.ID().Str(),
+			Date:      sn.Time.Format(TimeFormat),
+			Host:      sn.Hostname,
+			Tags:      sn.Tags,
+			Directory: sn.Paths}
+
+		response = append(response, k)
+	}
+
+	output, _ := json.Marshal(response)
```

Comment:
Use a `json.Decoder` here, like `dec := json.NewDecoder(os.Stdout)` and then use its `Decode()` method.